# HG changeset patch
# User Sam Gillingham <gillingham.sam@gmail.com>
# Date 1557888823 -36000
# Node ID 9762e1bfe7731adde47fe5fb4a8753798aaef2eb
# Parent  d2408fdef9c22e5247582a727e7ec17de8e7f11e
Fix testrios.py with GDAL 3 by ensuring correct extension used with KEA (this requirement was added by the /vsi changes in the KEA driver)

diff --git a/rios/riostests/teststats.py b/rios/riostests/teststats.py
--- a/rios/riostests/teststats.py
+++ b/rios/riostests/teststats.py
@@ -56,19 +56,19 @@
     # We repeat these tests on a number of different drivers, if they are available,
     # as some stats-related things may work fine on some drivers but not on others. 
     driverTestList = [
-        ('HFA', ['COMPRESS=YES']),
-        ('GTiff', ['COMPRESS=LZW', 'TILED=YES', 'INTERLEAVE=BAND']),
-        ('KEA', [])
+        ('HFA', ['COMPRESS=YES'], 'img'),
+        ('GTiff', ['COMPRESS=LZW', 'TILED=YES', 'INTERLEAVE=BAND'], 'tif'),
+        ('KEA', [], 'kea')
     ]
     # Remove any which current GDAL not suporting
-    driverTestList = [(drvrName, options) for (drvrName, options) in driverTestList
+    driverTestList = [(drvrName, options, ext) for (drvrName, options, ext) in driverTestList
         if gdal.GetDriverByName(drvrName) is not None]
     
     # Loop over all drivers
-    for (driverName, creationOptions) in driverTestList:
+    for (driverName, creationOptions, ext) in driverTestList:
         # Loop over all datatype tuples in the list
         for (fileDtype, arrDtype, scalefactor) in dataTypesList:
-            imgfile = 'test.img'
+            imgfile = 'test.' + ext
             ds = riostestutils.createTestFile(imgfile, dtype=fileDtype, driverName=driverName, 
                     creationOptions=creationOptions)
             rampArr = riostestutils.genRampArray().astype(arrDtype) * scalefactor
